dev-tools/set_version

```diff
@@ -1,38 +1,136 @@
 #!/usr/bin/env python
-import os
 import argparse
+import fileinput
+import os
+import re
+import sys
 from subprocess import check_call
 
-template_go = '''package beat
+pattern = re.compile(r'(const\s|)(\w*)(v|V)(ersion)(\s*)(:|=)(\s)(")(.*)(")(.*)')
+repl    = r'\1\2\3\4\5\6\7"{}"\11'
+
 
-const defaultBeatVersion = "{}"
+goversion_template = '''package main
+
+const appVersion = "{version}"
 '''
 
-template_packer = '''version: "{version}"
+goversion_template_libbeat = '''package beat
+
+const defaultBeatVersion = "{version}"
 '''
 
 
+yamlversion_template = '''version: "{version}"
+'''
+
+def split_path(path):
+    allparts = []
+    while 1:
+        parts = os.path.split(path)
+        if parts[0] == path:  # sentinel for absolute paths
+            allparts.insert(0, parts[0])
+            break
+        elif parts[1] == path: # sentinel for relative paths
+            allparts.insert(0, parts[1])
+            break
+        else:
+            path = parts[0]
+            allparts.insert(0, parts[1])
+    return allparts
+
+def get_substituted_content(filename, pattern, replacement, errormsg):
```

Comment:
But it would be simpler if we would mandate that the community beats must have a `version.go` file that we always overwrite, right? Is there any reason that wouldn't be a good idea?